src/QueryEditor.tsx

```diff
@@ -1,159 +1,38 @@
-import React from 'react';
-import { QueryEditorProps, SelectableValue } from '@grafana/data';
-import { DataSource } from './datasource';
-import { AthenaDataSourceOptions, AthenaQuery, defaultQuery, SelectableFormatOptions } from './types';
-import { InlineSegmentGroup } from '@grafana/ui';
+import React, { useCallback, useEffect, useState } from 'react';
 import { config } from '@grafana/runtime';
-import { FormatSelect, ResourceSelector } from '@grafana/aws-sdk';
-import { RunQueryButtons } from '@grafana/async-query-data';
-import { selectors } from 'tests/selectors';
-import { appendTemplateVariables } from 'utils';
-import SQLEditor from 'SQLEditor';
+import { QueryEditorProps } from '@grafana/data';
+import { AthenaDataSourceOptions, AthenaQuery } from './types';
+import { DataSource } from './datasource';
+import { QueryEditorForm } from './QueryEditorForm';
+import { QueryEditorHeader } from '@grafana/aws-sdk';
 
-type Props = QueryEditorProps<DataSource, AthenaQuery, AthenaDataSourceOptions> & {
-  hideOptions?: boolean;
-  hideRunQueryButtons?: boolean;
-};
+export function QueryEditor(props: QueryEditorProps<DataSource, AthenaQuery, AthenaDataSourceOptions>) {
+  const [dataIsStale, setDataIsStale] = useState(false);
+  const { onChange } = props;
 
-type QueryProperties = 'regions' | 'catalogs' | 'databases' | 'tables' | 'columns';
+  useEffect(() => {
+    setDataIsStale(false);
+  }, [props.data]);
```

Comment:
i tried adding a space and then deleting it and it lets me reenable the run query button.

Reply:
After talking to @kevinwcyu we decided not to go forward with comparing the queries, since it would require a deep comparison on every change and/or tracking changes, which might be an overkill for this feature.